backend/src/errors.rs

```diff
@@ -1,32 +1,47 @@
 use std::fmt;
-
-use axum::{http::StatusCode, response::IntoResponse, Json};
+use axum::{http::StatusCode,
+response::{IntoResponse, Response, Json}
+};
 use serde_json::json;
+use sqlx::Error as SqlxError;
 
 pub enum CustomError {
     BadRequest,
-    MessageNotFound,
-    ChannelNotFound,
-    UserNotFound(String),
+    NotFound(String),
+    DatabaseError(String),
     InternalServerError,
 }
 
+impl From<SqlxError> for CustomError {
+    fn from(err: SqlxError) -> Self {
+        match err {
+            SqlxError::RowNotFound => CustomError::NotFound("Entity not found".to_string()),
+            SqlxError::Database(e) => CustomError::DatabaseError(e.to_string()),
+            _ => CustomError::InternalServerError,
+        }
+    }
+}
+
```

Comment:
Looking at this again, RowNotFound seems redundant here, the Database error generally covers all db query errors. Could probably delete it. 

I think we need to think about what errors should be propegated to the user, which the signup errors do, like duplicate email/username when signing up or incorrect password when logging in. And handle the rest internally.

Reply:
I think the value in making it our own error is to make a custom error message to send to client.

Reply:
We might want to have different errors for each section of our app, so like `UserNotFound`, `MessageNotFound`, etc. This might be nice to do but also might not give us any advantage and just be a bunch of duplicate code

Reply:
At the least we can have a set of the common errors and make an enum of `UserError` sort of like this:
```
enum UserError {
  NotFoundError,
  DuplicateIdentifierError,
  ... Some other generic error used by other controllers
}
```

Reply:
@LukeBrandon since this works for now, should we just make error handling a separate issue?